File: 02-grid-traveller/01.py

def getWays(x, y):
    if x <= 0:
        return 0

    if y <= 0:
        return 0

    ways = 0

    # Up
    ways += getWays(x-1, y)

    # Left
    ways += getWays(x, y-1)

    # Reached the destination
    if x == y == 1:
        return 1

    return ways


if __name__ == '__main__':

    print(getWays(1, 0))
    print(getWays(0, 1))
    print(getWays(1, 1))
    print(getWays(3, 3))
    print(getWays(2, 2))

File: 02-grid-traveller/01.cpp
#include <iostream>

#define ll long long int

using namespace std;

ll getWays(int x, int y)
{
    if (x <= 0 || y <= 0)
        return 0;

    if (x == 1 && y == 1)
        return 1;

    ll ways = 0;
    ways += getWays(x - 1, y);
    ways += getWays(x, y - 1);

    return ways;
}

int main()
{

    cout << getWays(1, 1) << endl;
    cout << getWays(2, 3) << endl;
    cout << getWays(3, 2) << endl;
    cout << getWays(3, 3) << endl;
    cout << getWays(18, 18) << endl;

    return 0;
}